core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -1200,6 +1200,8 @@ private RESTViewBuilder(SessionContext context, TableIdentifier identifier) {
       checkViewIdentifierIsValid(identifier);
       this.identifier = identifier;
       this.context = context;
+      properties.putAll(
```

Comment:
I'm surprised that the same isn't done for tables. Can you please double-check whether default/override properties properly work with the rest catalog?

Reply:
in that case maybe we should set the default props here too, but it would be good to investigate why this isn't necessary for tables

Reply:
@ebyhr did you have a chance looking into this?

Reply:
@nastra  Perhaps, we need a similar change for tables. The new `CatalogTests.testDefaultTableProperties` doesn't pass without `RESTSessionCatalog` change. Added a new commit to fix it.